### PR TITLE
Update dataset failure distribution list

### PR DESCRIPTION
**Pull request by iaindillingham**
The distribution list contains the @names of users who should be notified on dataset failure. With @LiamHart-hub's agreement, we will make the following changes:

* Remove @Jongmassey
* Add @JulietUnderdown1
* Remove @inglesp 
* Add @iaindillingham, to help Juliet understand future dataset failures

### CODE REVIEW

**Review by Jongmassey — APPROVED** [review]
👍 

Could you explain a little more about `codelists.json` and why those failures hadn't happened/caused problems before?

Also, IIRC there might be some deploy faff for this once merged.

**Comment by iaindillingham** [comment]
I'm afraid I didn't dig into why `codelists.json` is required now, but wasn't required when the repo was last pushed (2022-12-22). However, that's over a year ago and a lot has changed since then. You may be interested in opensafely/gh-testing-research#4, which caused some of the earlier failures.

**Review by lucyb — APPROVED** [review]
Thank you for explaining _why_ each change was being made.

As mentioned earlier, I think we should review this new list of people in a few weeks and change it again if necessary.

**Pull request merged**
